homeassistant/components/tfl/sensor.py

```diff
@@ -0,0 +1,160 @@
+                    stop_point_api,
+                    stop_point_infos["commonName"],
+                    stop_point_ids[0],
+                    entry.entry_id,
+                )
+            )
+
+        async_add_entities(devices, True)
+
+    except HTTPError as exception:
+        error_code = exception.code
+        _LOGGER.exception(
+            "Error retrieving stop point data from TfL for stop_points=%s with HTTP error_code=%s, entities will not be created",
+            stop_point_ids,
+            error_code,
+        )
+    except URLError as exception:
+        _LOGGER.exception(
+            "Error retrieving stop point data from TfL for stop_points=%s with URLError reason=%s, entities will not be created",
+            stop_point_ids,
+            exception.reason,
+        )
+
+
+class StopPointSensor(SensorEntity):
+    """Representation of a TfL StopPoint as a Sensor."""
+
+    _attr_attribution = "Powered by TfL Open Data"
+    _attr_icon = "mdi:bus"
+
+    def __init__(
+        self, stop_point_api: stopPoint, name: str, stop_point_id: str, entry_id: str
+    ) -> None:
+        """Initialize the TfL StopPoint sensor."""
+        # super().__init__(coordinator)
+        self._name = name
+        self._attr_name = name
+        self._attr_unique_id = stop_point_id
+        self._attr_device_info = DeviceInfo(
+            name="TfL",
+            identifiers={(DOMAIN, entry_id)},
+            entry_type=DeviceEntryType.SERVICE,
+        )
+
+        self._stop_point_api = stop_point_api
+        self._stop_point_id = stop_point_id
+
+    async def async_update(self) -> None:
+        """Update Stop Point state."""
+        try:
+            attributes = {}
+
+            def raw_arrival_to_arrival_mapper(raw_arrival):
+                return {
+                    "line_name": raw_arrival[RAW_ARRIVAL_LINE_NAME],
+                    "destination_name": raw_arrival[RAW_ARRIVAL_DESTINATION_NAME],
+                    "time_to_station": raw_arrival[RAW_ARRIVAL_TIME_TO_STATION],
+                }
+
+            raw_arrivals = await call_tfl_api(
+                self.hass, self._stop_point_api.getStationArrivals, self._stop_point_id
+            )
+
+            if raw_arrivals:
+                raw_arrivals_sorted = sorted(
+                    raw_arrivals, key=itemgetter(RAW_ARRIVAL_TIME_TO_STATION)
+                )
+
+                arrivals = list(map(raw_arrival_to_arrival_mapper, raw_arrivals_sorted))
+                _LOGGER.debug("Got arrivals=%s", arrivals)
+
+                arrival_next = arrivals[0]
+                arrivals_next_3 = arrivals[:3]
+
+                # Due to 255 character limit, the value of the sensor is the next arrival and
+                # the next 3 and full list are provided as attributes
+                self._attr_native_value = arrival_next
+                attributes[ATTR_NEXT_THREE_ARRIVALS] = arrivals_next_3
+                attributes[ATTR_NEXT_ARRIVALS] = arrivals
```

Comment:
Mostly, you'd be interested in showing the next 3 arrivals at a stop rather than everything, which could be say 12 arrivals.
So as a convenience, it seemed nice to provide an attribute explicitly for the next 3 arrivals so that usages in markdown for example on dashboards don't have to only take the top 3 items and can instead just iterate over the list knowing that it won't be enormous.
Should make life simpler to use it

Reply:
Recommendation from @joostlek to use a coordinator with multiple sensors for each arrival as per `swiss_public_transport` which would result in how this looks changing